src/main/flight/gps_rescue.c

```diff
@@ -394,22 +394,30 @@ static void performSanityChecks(void)
         previousDistanceToHomeCm = rescueState.sensor.distanceToHomeCm;
         secondsLowSats = 0;
         secondsDoingNothing = 0;
-        return;
     }
 
     // Handle events that set a failure mode to other than healthy.
     // Disarm via Abort when sanity on, or for hard Rx loss in FS_ONLY mode
     // Otherwise allow 20s of semi-controlled descent with impact disarm detection
-    const bool hardFailsafe = !rxIsReceivingSignal();
     if (rescueState.failure != RESCUE_HEALTHY) {
-        if (gpsRescueConfig()->sanityChecks == RESCUE_SANITY_ON) {
-            rescueState.phase = RESCUE_ABORT;
-        } else if ((gpsRescueConfig()->sanityChecks == RESCUE_SANITY_FS_ONLY) && hardFailsafe) {
+        switch(gpsRescueConfig()->sanityChecks) {
+        case RESCUE_SANITY_ON:
             rescueState.phase = RESCUE_ABORT;
-        } else {
+            break;
+        case RESCUE_SANITY_FS_ONLY:
+            if (!rxIsReceivingSignal()) {
+                rescueState.phase = RESCUE_ABORT;
+            }
+            break;
+        default:
             // even with sanity checks off, 
             rescueState.phase = RESCUE_DO_NOTHING; // 20s semi-controlled descent with impact detection, then abort
         }
+
+        // override when Allow Arming without Fix is enabled and there is no GPS_FIX_HOME and we are not testing with FAILSAFE mode.
+        if (gpsRescueConfig()->allowArmingWithoutFix && !STATE(GPS_FIX_HOME) && !FLIGHT_MODE(FAILSAFE_MODE)) {
+            rescueState.phase = RESCUE_ABORT;
+        }
```

Comment:
When `RESCUE_SANITY_OFF` is enabled this disables `RESCUE_DO_NOTHING` if `allowArmingWithoutFix` is enabled and not `fail-safe` mode switch induced to aid in testing. Note this condition fails if home position is acquired.

Reply:
Other edge cases should be addressed in a new PR as this PR only fixes the sanity validation and now adds override when sanity was disabled by user in combination with `allowArmingWithoutFix`.